clippy_lints/src/iter_without_into_iter.rs

```diff
@@ -56,7 +102,77 @@ fn is_nameable_in_impl_trait(ty: &rustc_hir::Ty<'_>) -> bool {
     !matches!(ty.kind, TyKind::OpaqueDef(..))
 }
 
+fn type_has_inherent_method(cx: &LateContext<'_>, ty: Ty<'_>, method_name: Symbol) -> bool {
+    if let Some(ty_did) = ty.ty_adt_def().map(ty::AdtDef::did) {
+        cx.tcx.inherent_impls(ty_did).iter().any(|did| {
+            cx.tcx
+                .hir()
+                .expect_item(did.expect_local())
```

Comment:
Might be better to return `None` for non-local types. The current orphan rules wouldn't allow it, but those could change in the future.

Reply:
Probably don't need this code if we're going with `associated_items`, since we can just give that a normal DefId

Reply:
It will at least not be an ICE. Good enough given such a change is unlikely.